platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/IdpMetadata.java

```diff
@@ -53,18 +53,18 @@
 
     private String encryptionCertificate;
 
-    private Map<String, EntityDescriptor> entryDescriptions = new ConcurrentHashMap<>();
+    private String metadata;
+
+    private AtomicReference<Map<String, EntityDescriptor>> entryDescriptions =
+            new AtomicReference<>();
 
     private String singleLogoutBinding;
 
     private String singleLogoutLocation;
 
-    public void setMetadata(String metadata)
-            throws WSSecurityException, XMLStreamException, SAMLException, IOException {
-        MetadataConfigurationParser metadataConfigurationParser = new MetadataConfigurationParser(
-                Collections.singletonList(metadata),
-                ed -> entryDescriptions.put(ed.getEntityID(), ed));
-        entryDescriptions.putAll(metadataConfigurationParser.getEntryDescriptions());
+    public void setMetadata(String metadata) {
+        this.metadata = metadata;
+        entryDescriptions.set(null);
```

Comment:
:question: Why instantiate this to null? By default `new AtomicReference<>();` is going to have a null reference already.